Index: C/cel.c
===================================================================
--- C/cel.c
+++ C/cel.c
@@ -243,6 +243,8 @@
           latp = -90.0;
         }

+         u = 0.0;
+         v = 0.0;
        } else {
          slz = slat0/z;
          if (fabs(slz) > 1.0) {
Index: C/tab.c
===================================================================
--- C/tab.c
+++ C/tab.c
@@ -1081,10 +1081,13 @@
   if (M > 1) {
     nv = 1 << M;
     tabcoord = calloc(nv, sizeof(double *));
+  } else {
+    nv = 0;
+    tabcoord = 0x0;
   }


   status = 0;
   wp = world;
Index: C/wcshdr.c
===================================================================
--- C/wcshdr.c
+++ C/wcshdr.c
@@ -506,6 +506,7 @@
     col0 = colax[0];
   } else {
     primage = 1;
+    col0 = 0;
   }


